### PR TITLE
Model testing

### PR DESCRIPTION
**Pull request by MichaelChan20**
Added non-determinism, memory usage and data slice (http vs https urls) tests.
These tests are manual only as they require the dvc output files.

Data slice test is marked as manual.

Non-determinism and memory usage are marked as training as they train a model and therefore take very long. (30min)

Mutamorphic testing is WIP (not in this PR) and would need a manual run as well.

Some of the manual tests would be able to run in the pipeline if we ran dvc repro beforehand, however this would take 15 mins.
Memory usage and non-determinism test could be combined or planned in such a way it would only require another 15mins for training one extra model.

### CODE REVIEW

File: tests/test_model_development.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    #First model (pipeline output)
+    trained_model1 = tf.keras.models.load_model(os.path.join(path, "models", "trained_model.keras"))
+    predictions_model1 = predict_classes(trained_model1, X_test)
+    results_model1 = evaluate_results(y_test, predictions_model1)
+    
+    
+    #Set seeds
+    random.seed(694201337)
+    np.random.seed(694201337)
+    tf.random.set_seed(694201337)
```
Comment (janvandermeulen) [inline tests/test_model_development.py:76]: Maybe define this as a variable?

**Review by janvandermeulen — APPROVED** [review]
I reviewed it over and it looks good to me, nice work with doing the other 2 angles of the tests. Let's not do the PR an hour before the deadline next week. For next week we will probably need to fix the pipeline again.

File: tests/test_model_development.py (hunk 1 above)
Comment (MichaelChan20) [reply tests/test_model_development.py:76]: Will add this once we figure out the pipeline

File: tests/test_preprocess.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def test_char_index():
         char_index = json.load(file)
     assert len(char_index) == 53
     
-@pytest.mark.slow
+@pytest.mark.manual
 def test_real_data():
     sys.argv = ["", "data"]
     preprocess.main()
```
Comment (janvandermeulen) [inline tests/test_preprocess.py:43]: Tokenizer.pkl seems to be missing for this file in the pipeline.

**Pull request merged**